pkgs/development/python-modules/embedchain/default.nix

```diff
@@ -0,0 +1,105 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+
+  # build-system
+  poetry-core,
+
+  # dependencies
+  python-dotenv,
+  langchain,
+  requests,
+  openai,
+  chromadb,
+  posthog,
+  rich,
+  beautifulsoup4,
+  pypdf,
+  gptcache,
+  pysbd,
+  mem0ai,
+  schema,
+  langchain-openai,
+  sqlalchemy,
+  alembic,
+  langchain-cohere,
+  langchain-community,
+  langsmith,
+
+  # runtime dependencies
+  google-cloud-aiplatform,
+}:
+
+let
+  version = "0.1.123";
+  repo = fetchFromGitHub {
+    owner = "mem0ai";
+    repo = "mem0";
+    rev = "refs/tags/${version}";
+    hash = "sha256-/+A2w1HWq2toZOyCMd580p/Nd0rXs1w2eDD5Y3t3nP4=";
+  };
+in
+buildPythonPackage {
+  pname = "embedchain";
+  inherit version;
+  pyproject = true;
+
+  disabled = pythonOlder "3.9";
+
+  nativeBuildInputs = [ poetry-core ];
+
+  src = "${repo}/embedchain";
```

Comment:
I think there's a better way to do that, I don't remember it at the moment, maybe @GaetanLepage knows ?

Reply:
It's `build-system = [ poetry-core ];` instead.

Reply:
Switch `repo` to `src` (i.e. move it back into `buildPythonPackage`) then use `sourceRoot = "${src.name}/embedchain";`